### PR TITLE
add plugin thanos

### LINKED ISSUE #71: Provide production prometheus federation and long term storage capability
**What would you like to be added**:

Previously we support prometheus federation, which is a basic model that can do multicluster metrics aggregate, but far from production ready.

I have investigated recently, and this one https://elastisys.com/long-term-metrics-storage-thanos-vs-cortex-vs-influxdb-vs-m3db-vs-victoriametrics-vs-timescaledb/ is the most complete comparison with clear criteria.

The conclusion is [thanos ](https://thanos.io/) is the winner for long time storage of multicluster metrics. 

As we can see thanos some some dependencies https://thanos.io/tip/thanos/getting-started.md/#dependencies Particularly we should pay attention to `object storage`, which we have not supported in kurator.

@daixiang0, for the s3 object storage, can we use [minio](https://github.com/minio/minio) to provide rather than using AWS directly?

### PR DESCRIPTION
**Pull request by zirain**
**What type of PR is this?**

/kind feature

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #71 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
add plugin thanos
```

### CODE REVIEW

File: docs/content/en/docs/Components/thanos.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+### What is Thanos
+
+[Thanos](https://thanos.io) provides a global query view, high availability, data backup with historical, cheap data access as its core features in a single binary.
+
+In this task, we will show you how to install Thanos with kurator.
+
+### Prerequisites
+
+This task requires you have installed karmada and have joined at least one member cluster.
```
Comment (zirain) [inline docs/content/en/docs/Components/thanos.md:16]: should be updated after https://github.com/kurator-dev/kurator/pull/78#issuecomment-1253282709 merged

File: Makefile [hunk 2]
```diff
@@ -61,13 +66,34 @@ golangci-lint:
...
+	rm -rf ${PROM_THANOS_OUT_PATH}
+	rm -rf ${PROM_THANOS_MANIFESTS_PATH}
+	mkdir -p ${PROM_THANOS_MANIFESTS_PATH}
+	mkdir -p ${PROM_THANOS_OUT_PATH}
+	cp manifests/jsonnet/prometheus/thanos.jsonnet ${PROM_THANOS_OUT_PATH}/kube-prometheus.jsonnet
+	hack/gen-prom.sh ${PROM_THANOS_OUT_PATH} ${KUBE_PROM_VER} kube-prometheus.jsonnet
+	cp -r ${PROM_THANOS_OUT_PATH}/manifests/* ${PROM_THANOS_MANIFESTS_PATH}
+
+.PHONY: gen-thanos
+gen-thanos:
```
Comment (hzxuzhonghu) [inline Makefile:88]: Suggest moving component gen out, in this root makefile we only need to include that specific components files

Comment (zirain) [reply Makefile:88]: will do that in separate PR, same to prometheus, let's track this by issue

**Review by hzxuzhonghu — COMMENTED** [review]
LGTM @Xieql  please have a look at the docs

**Review by hzxuzhonghu — APPROVED** [review]

**Pull request merged**